models/multiphase/d3q27_pf_velocity/Dynamics.c.Rt

```diff
@@ -938,77 +1009,164 @@ CudaDeviceFunction void ENeumann(){
 		}
 	?>
 }
-#endif
-
-CudaDeviceFunction void WVelocity(){
-	U = VelocityX;
-	V = VelocityY;
-	W = VelocityZ;
-	if ( developedFlow > 0.1 ){
-		U = 6.0 * Uavg * Y*(HEIGHT - Y)/(HEIGHT*HEIGHT);
-		V = 0.0;
-		W = 0.0;
-	}
-	if ( developedPipeFlow_X > 0.1 ){
-		U = 2.0 * Uavg * (1 - pow( (sqrt(pow((Y-pipeCentre_Y),2) + pow((Z-pipeCentre_Z),2)) / pipeRadius),2));
-		V = 0.0;
-		W = 0.0;
-	}
+CudaDeviceFunction void WNeumann(){
 	<?R
-		n = c(-1,0,0)
-		geq = EQ$feq
-		bounce = Bounce(U)
-		sel = as.vector( (U %*% n) < 0)
-
-		sel2 = as.vector( ( U %*% n) == 0)	
-		exM = (g[sel2] - geq[sel2]) %*% U[sel2,]
-        Unknowns = 1.0/max(colSums(U[sel2,]^2)) # should be 6 for q27, 4 for q15
-        Nmat = t(U[sel,]) %*% U[sel,]
-
-		C(g[sel], g[bounce][sel] + (geq[sel] - geq[bounce][sel]) - 0.5* (exM %*% solve(Nmat) %*% t(U))[sel])
+		if (Options$OutFlow){
+			C(g, PV(paste0(g_neighbours,"(",-Density$dx[Density$group=="g"]+1,",",-Density$dy[Density$group=="g"],",",-Density$dz[Density$group=="g"],")")))	
+			C(h, PV(paste0(h_neighbours,"(",-Density$dx[Density$group=="h"]+1,",",-Density$dy[Density$group=="h"],",",-Density$dz[Density$group=="h"],")")))	
+		}
 	?>
-	<?R
-		U_PF = U[1:PF_velocities,]
-        	pf = PV("PhaseField")
-		n = c(-1,0,0)
-		heq = pf * EQ_h$feq
-		bounce = Bounce(U_PF)
-		sel = as.vector( (U_PF %*% n) < 0)
+}
 
-		sel2 = as.vector( ( U_PF %*% n) == 0)	
-		exM = (h[sel2] - heq[sel2]) %*% rep(1,sum(sel2)) 
-		Unknowns = 1.0/sum(sel)
+/* If needed, user can uncomment and adjust code for these B.C.
```

Comment:
I think you should be able to generate those with R as well, similarly as you do for the pressure/velocity boundaries. Otherwise there is quite a lot of duplicated and commented out code, which is not nice